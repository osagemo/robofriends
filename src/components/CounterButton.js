import React, { Component } from "react";

export default class CounterButton extends Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0
    };
  }

  shouldComponentUpdate(nextProps, nextState) {
    if (this.state.count !== nextState.count) {
      return true;
    } else {
      return false;
    }
  }

  updateCount = () => {
    this.setState(state => {
      return { count: state.count + 1 };
    });
  };
  render() {
    return (
      <div>
        <button color={this.props.color} onClick={this.updateCount}>
          Count: {this.state.count}
        </button>
      </div>
    );
  }
}
