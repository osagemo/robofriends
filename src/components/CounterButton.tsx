import React, { Component } from "react";

interface ICounterButtonProps {}

interface ICounterButtonState {
  count: number;
}

export default class CounterButton extends Component<
  ICounterButtonProps,
  ICounterButtonState
> {
  constructor(props: ICounterButtonProps) {
    super(props);
    this.state = {
      count: 0
    };
  }

  shouldComponentUpdate(nextState: ICounterButtonState) {
    if (this.state.count !== nextState.count) {
      return true;
    } else {
      return false;
    }
  }

  updateCount = () => {
    this.setState(state => {
      return { count: state.count + 1 };
    });
  };
  render() {
    return (
      <div>
        <button id="counter" onClick={this.updateCount}>
          Count: {this.state.count}
        </button>
      </div>
    );
  }
}
