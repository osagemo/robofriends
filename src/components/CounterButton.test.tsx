import { shallow, mount } from "enzyme";
import React from "react";
import CounterButton from "./CounterButton";

it("Renders CounterButton component", () => {
  const wrapper = mount(<CounterButton />);
  expect(wrapper).toMatchSnapshot();
});

it("Only updates when count of nextState is different", () => {
  const wrapper = shallow(<CounterButton />);
  wrapper.setState({ count: 2 });
  let shouldUpdate = count =>
    wrapper.instance().shouldComponentUpdate({ count });

  expect(shouldUpdate(2)).toBe(false);
  expect(shouldUpdate(3)).toBe(true);
});

it("Updates count when clicking the button", () => {
  const wrapper = shallow(<CounterButton />);
  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.instance().state.count).toBe(1);
  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.instance().state.count).toBe(2);
});

it("Recieves props correctly", () => {
  const wrapper = shallow(<CounterButton color={"red"} />);
  expect(wrapper.instance().props.color).toBe("red");
});
