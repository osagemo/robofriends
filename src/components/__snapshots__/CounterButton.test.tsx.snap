// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CounterButton component 1`] = `
<CounterButton>
  <div>
    <button
      id="counter"
      onClick={[Function]}
    >
      Count: 
      0
    </button>
  </div>
</CounterButton>
`;
